modules/homebrew.nix

```
@@ -1,219 +1,751 @@
+        ++ optional (config.cleanup == "zap") "--zap"
+      );
+    };
+  };
+
+  globalOptions = { config, ... }: {
+    options = {
+      brewfile = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable Homebrew to automatically use the Brewfile that this module generates in
+          the Nix store, when you manually invoke <command>brew bundle</command>.
+
+          Implementation note: when enabled, this option sets the
+          <literal>HOMEBREW_BUNDLE_FILE</literal> environment variable to the path of the Brewfile
+          that this module generates in the Nix store, by adding it to
+          ${mkDocOptionLink "environment.variables"}.
+        '';
+      };
+      noAutoUpdate = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to disable Homebrew from automatically updating itself and all formulae
+          when you manually invoke commands like <command>brew install</command>,
+          <command>brew upgrade</command>, <command>brew tap</command>, and
+          <command>brew bundle [install]</command>.
+
+          You may want to consider enabling this option if you have
+          ${mkDocOptionLink "homebrew.onActivation.upgrade"} enabled, and
+          ${mkDocOptionLink "homebrew.onActivation.update"} disabled, if you want to ensure that
+          your installed formulae will only be upgraded during <command>nix-darwin</command> system
+          activation, after you've explicitly run <command>brew update</command>.
+
+          Implementation note: when enabled, this option sets the
+          <literal>HOMEBREW_NO_AUTO_UPDATE</literal> environment variable, by adding it to
+          ${mkDocOptionLink "environment.variables"}.
+        '';
+      };
+      noLock = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to disable lockfile generation when you manually invoke
+          <command>brew bundle [install]</command>. This is often desirable when
+          ${mkDocOptionLink "homebrew.global.brewfile"} is enabled, since
+          <command>brew bundle [install]</command> will try to write the lockfile in the Nix store,
+          and complain that it can't (though the command will run successfully regardless).
+
+          Implementation note: when enabled, this option sets the
+          <literal>HOMEBREW_BUNDLE_NO_LOCK</literal> environment variable, by adding it to
+          ${mkDocOptionLink "environment.variables"}.
+        '';
+      };
+
+      homebrewEnvironmentVariables = mkInternalOption { type = types.attrs; };
+    };
+
+    config = {
+      homebrewEnvironmentVariables = {
+        HOMEBREW_BUNDLE_FILE = mkIf config.brewfile "${brewfileFile}";
+        HOMEBREW_BUNDLE_NO_LOCK = mkIf config.noLock "1";
+        HOMEBREW_NO_AUTO_UPDATE = mkIf config.noAutoUpdate "1";
```

```suggestion
        HOMEBREW_BUNDLE_NO_LOCK = mkIf (!config.lockfile) "1";
        HOMEBREW_NO_AUTO_UPDATE = mkIf (!config.autoUpdate) "1";
```

Accompanying changes for earlier suggestions.